### ISSUE TITLE
Breeze asset compilation causing OOM on dev-mode

### ISSUE THREAD

**Issue by pierrejeambrun** [apache/airflow#31384]
### Apache Airflow version

main (development)

### What happened

asset compilation background thread is not killed when running `stop_airflow`  or `breeze stop`.

Webpack process takes a lot of memory, each `start-airflow` starts 4-5 of them.

After a few breeze start, we end up with 15+ webpack background processes that takes more than 20G RAM

### What you think should happen instead

`run_compile_www_assets` should stop when running `stop_airflow` from tmux. It looks like it spawn a `compile-www-assets-dev` pre-commit in a subprocess, that doesn't get killed when stopping breeze

### How to reproduce

```
breeze start-airflow --backend postgres --python 3.8 --dev-mode
# Wait for tmux session to start
breeze_stop

breeze start-airflow --backend postgres --python 3.8 --dev-mode
# Wait for tmux session to start
breeze_stop

# do a couple more if needed
```

Open tmux and monitor your memory, and specifically webpack processes.

### Operating System

Ubuntu 20.04.6 LTS

### Deployment

Other

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by pierrejeambrun** [comment]
I looked a bit and didn't find a immediate fix for it. (I am not really familiar with breeze code). I will take a closer look once I am done with the dashboard PR.

**Comment by potiuk** [comment]
See the description (and fix) in #31403) - it was the abrupt stopping of daemon Thread that caused it (and replacing it with forking, creating a new process group and sending SIGTERM to the process group atexit seems to do the job).
